src/function/table/arrow_conversion.cpp

```diff
@@ -612,6 +613,10 @@ void ColumnArrowToDuckDB(Vector &vector, ArrowArray &array, ArrowScanLocalState
 			ArrowToDuckDBMapList(*child_entries[type_idx], *struct_arrow.children[type_idx], scan_state, size,
 			                     arrow_convert_data, col_idx, arrow_convert_idx, offsets, &struct_validity_mask);
 		}
+		// TODO remove this when arrow ensures keys are unique
+		if (!AreKeysUnique(*child_entries[0], size)) {
```

Comment:
Should we not check null-ness here as well? Perhaps we can combine this into a single function`IsValidMap` that returns an `enum class MapInvalidReason { MAP_VALID, MAP_CONTAINS_NULL, MAP_CONTAINS_DUPLICATE_KEY };`?

Reply:
Can we add this function to `Vector::Verify` as well?

Reply:
replaced these with one VerifyMap function, also added it to Vector::Verify 👍